src/PaginationResult.php

```diff
@@ -50,10 +50,20 @@ public function __construct($rows, array $meta)
     /**
      * Get iterator of records.
      *
-     * @return \ArrayIterator
+     * @return \Traversable|\ArrayIterator
      */
     public function getIterator()
     {
-        return new \ArrayIterator($this->records);
+        return $this->records instanceof \Traversable ? $this->records : new \ArrayIterator($this->records);
+    }
+
+    /**
+     * Count records.
+     *
+     * @return int
+     */
+    public function count()
+    {
+        return $this->records instanceof \Countable || is_array($this->records) ? count($this->records) : 0;
```

Comment:
Why does it return `0` when value is unavailable? `count` basically returns 1 when it is not countable.

Reply:
Is this behavior still available in next generation PHP?

Reply:
[PHP: rfc:counting_non_countables](https://wiki.php.net/rfc/counting_non_countables)

Reply:
Should we return `1` triggering warning?

Reply:
I think so.